lib/ApiRequestor.php

```diff
@@ -164,31 +164,32 @@ private static function _specificAPIError($rbody, $rcode, $rheaders, $resp, $err
         $param = isset($errorData['param']) ? $errorData['param'] : null;
         $code = isset($errorData['code']) ? $errorData['code'] : null;
         $type = isset($errorData['type']) ? $errorData['type'] : null;
+        $declineCode = isset($errorData['decline_code']) ? $errorData['decline_code'] : null;
 
         switch ($rcode) {
             case 400:
                 // 'rate_limit' code is deprecated, but left here for backwards compatibility
                 // for API versions earlier than 2015-09-08
                 if ($code == 'rate_limit') {
-                    return new Error\RateLimit($msg, $param, $rcode, $rbody, $resp, $rheaders);
+                    return new Error\RateLimit($msg, $param, $rcode, $rbody, $resp, $rheaders, $code);
                 }
                 if ($type == 'idempotency_error') {
-                    return new Error\Idempotency($msg, $rcode, $rbody, $resp, $rheaders);
+                    return new Error\Idempotency($msg, $rcode, $rbody, $resp, $rheaders, $code);
                 }
 
                 // intentional fall-through
             case 404:
-                return new Error\InvalidRequest($msg, $param, $rcode, $rbody, $resp, $rheaders);
+                return new Error\InvalidRequest($msg, $param, $rcode, $rbody, $resp, $rheaders, $code);
             case 401:
-                return new Error\Authentication($msg, $rcode, $rbody, $resp, $rheaders);
+                return new Error\Authentication($msg, $rcode, $rbody, $resp, $rheaders, $code);
             case 402:
-                return new Error\Card($msg, $param, $code, $rcode, $rbody, $resp, $rheaders);
+                return new Error\Card($msg, $param, $code, $rcode, $rbody, $resp, $rheaders, $declineCode);
```

Comment:
Why does this pass `declineCode` and all the others pass `$code`? I don't fully grasp what you are trying to do here to be honest.

Reply:
`Error\Card` was the only class that already accepted the `code` parameter (as its 3rd argument).

When we introduced error codes for non-card errors, we didn't want to modify the constructors so we extracted the error code from the JSON body (cf. https://github.com/stripe/stripe-php/pull/559/files#diff-600ddc684b71253491fd6da75bb5f470L25).

`Error\Card` did something similar for decline codes (cf. https://github.com/stripe/stripe-php/pull/559/files#diff-0d197aceb8c08f2f4ddf6a9b6d0a0a68L28).

This PR updates all constructors so that both code (for all errors) and decline codes (for card errrors) are passed explicitly as constructor arguments.